lib/installer.js

```diff
@@ -1,35 +1,57 @@
-import request from 'request-promise';
-import { system, fs } from 'appium-support';
+import _ from 'lodash';
+import { system, fs, util, timing } from 'appium-support';
 import path from 'path';
 import { exec } from 'teen_process';
 import log from './logger';
+import B from 'bluebird';
+import os from 'os';
+import axios from 'axios';
+
 
 const WAD_VER = '1.2-RC';
 const WAD_DL = `https://github.com/Microsoft/WinAppDriver/releases/download/v${WAD_VER}/WindowsApplicationDriver.msi`;
 const WAD_DL_MD5 = 'dbaa9a3f7416c2b73cc5cd0e7452c8d0';
+const WAD_TIMEOUT = 5000; //ms
 
-let WAD_INSTALL_PATH = process.env['ProgramFiles(x86)'] || process.env.ProgramFiles || 'C:\\Program Files';
-WAD_INSTALL_PATH = path.resolve(WAD_INSTALL_PATH, 'Windows Application Driver',
-                                'WinAppDriver.exe');
+const WAD_INSTALL_PATH = path.resolve(
+  process.env['ProgramFiles(x86)'] || process.env.ProgramFiles || 'C:\\\\Program Files',
+  'Windows Application Driver', 'WinAppDriver.exe');
 const WAD_EXE_MD5 = '50d694ebfaa622ef7e4061c1bf52efe6';
 const WAD_GUID = 'DDCD58BF-37CF-4758-A15E-A60E7CF20E41';
 
+
+async function downloadFile (url, dstPath) {
+  const writer = fs.createWriteStream(dstPath);
+  const response = await axios({
+    url,
+    method: 'GET',
+    responseType: 'stream',
+    timeout: WAD_TIMEOUT,
+  });
+  response.data.pipe(writer);
+
+  await new B((resolve, reject) => {
+    writer.on('finish', resolve);
+    writer.on('error', reject);
+  });
+}
+
 async function downloadWAD () {
-  log.info(`Opening temp file to write WinAppDriver to...`);
-  let tempFile = path.resolve(process.env.TEMP, 'WinAppDriver.msi');
-  log.info(`Will write to ${tempFile}`);
+  const tempFile = path.resolve(os.tmpdir(), `${util.uuidV4()}.msi`);
```

Comment:
Is `tempDir` in appium-support better?

Reply:
nah, internally it uses the same API

The call to tempDir returns not just a name, but also creates a dir with random name in TMP folder. I don't need that and but only a path